gourmet/keymanager.py

```diff
@@ -142,24 +143,26 @@ def look_for_key (self, txt):
         The higher the score, the more probable the match.
         """
 
-        txt = txt.lower()
-        retvals = {}
+        txt = txt.casefold()
+        retvals = defaultdict(float)
         # First look for matches for our full text (or full text +/- s
         main_txts = [txt]
         main_txts.extend(defaults.guess_singulars(txt))
         if len(main_txts)==1:
             main_txts.extend(defaults.guess_plurals(txt))
         for t in main_txts:
-            is_key = self.rm.fetch_one(self.rm.ingredients_table,ingkey=t)
-            if is_key>=0:
+            # Look up for an entry in the ingredient table where the term t is
+            # found in the value from the ingkey column.
+            # By doing so, it establishes a baseline for the accuracy of t
+            # being a useful keyword.
+            entry = self.rm.fetch_one(self.rm.ingredients_table, ingkey=t)
+            if entry is not None:
                 retvals[t]=.9
             exact = self.rm.fetch_all(self.rm.keylookup_table,
                                       item=t)
             if exact:
```

Comment:
Does exact always return something iterable or sometimes `None`? Maybe we can remove the `if` as well.

Reply:
Yes, `exact` is always a `list`, the if statement is redundant.